k8s/src/main/scala/io/buoyant/k8s/Ns.scala

```diff
@@ -0,0 +1,93 @@
+package io.buoyant.k8s
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.service.Backoff
+import com.twitter.finagle.tracing.Trace
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util._
+
+abstract class Ns[O <: KubeObject: Manifest, W <: Watch[O]: Manifest, L <: KubeList[O]: Manifest, Cache](
+  backoff: Stream[Duration] = Backoff.exponentialJittered(10.milliseconds, 10.seconds),
+  timer: Timer = DefaultTimer.twitter
+) {
+  // note that caches must be updated with synchronized
+  private[this] val caches = Var[Map[String, Cache]](Map.empty[String, Cache])
```

Comment:
It seems like we actually have a pretty well-defined for interface for Cache.  Does it make more sense to just have NsCache and ServiceCache implement a common interface that handles initialize, update, etc?